From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: cyrozap <cyrozap@gmail.com>
Date: Mon, 19 Jan 2026 17:12:26 -0600
Subject: [PATCH] 8891: Fix lexing of directives with quoted strings in C
 headers

With this change, importing C header files containing lines like
`# if __has_include ("linux/stat.h")` will no longer cause lexer errors.

Fixes #7032
---
 .../Base/src/main/javacc/ghidra/app/util/cparser/CPP/CPP.jj    | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Ghidra/Features/Base/src/main/javacc/ghidra/app/util/cparser/CPP/CPP.jj b/Ghidra/Features/Base/src/main/javacc/ghidra/app/util/cparser/CPP/CPP.jj
index c069535c1f..a4cd68033d 100644
--- a/Ghidra/Features/Base/src/main/javacc/ghidra/app/util/cparser/CPP/CPP.jj
+++ b/Ghidra/Features/Base/src/main/javacc/ghidra/app/util/cparser/CPP/CPP.jj
@@ -2699,7 +2699,8 @@ SKIP : {
     <_INCWSP: <WSP> > : INCDEF |
     <_INCCP: <CP> > : DIRECTIVE |
     <_INCOP: <OP> > : INCDEF |
-    <_INCSTANDARD:  <_LT> (~["\n","\r",">",")","("])* <_GT> > : INCDEF
+    <_INCSTANDARD:  <_LT> (~["\n","\r",">",")","("])* <_GT> > : INCDEF |
+    <_INCSTRING: <QUOTED_TEXT> > : INCDEF
 }
 
 //<INCDEF>
-- 
2.45.1

